datafusion/src/physical_plan/expressions/binary.rs

```diff
@@ -298,12 +304,12 @@ fn evaluate_inverse_scalar(
 ) -> Result<Option<Arc<dyn Array>>> {
     use Operator::*;
     match op {
-        Lt => evaluate_scalar(rhs, &GtEq, lhs),
-        Gt => evaluate_scalar(rhs, &LtEq, lhs),
-        GtEq => evaluate_scalar(rhs, &Lt, lhs),
-        LtEq => evaluate_scalar(rhs, &Gt, lhs),
-        Eq => evaluate_scalar(rhs, &NotEq, lhs),
-        NotEq => evaluate_scalar(rhs, &Eq, lhs),
+        Lt => evaluate_scalar(rhs, &Gt, lhs),
+        Gt => evaluate_scalar(rhs, &Lt, lhs),
+        GtEq => evaluate_scalar(rhs, &LtEq, lhs),
+        LtEq => evaluate_scalar(rhs, &GtEq, lhs),
+        Eq => evaluate_scalar(rhs, &Eq, lhs),
+        NotEq => evaluate_scalar(rhs, &NotEq, lhs),
```

Comment:
A bug here because of the semantic of `evaluate_inverse_scalar`   i.e.   "5 < a"  should be transformed to " a > 5"

Reply:
haha, good catch!